thunder/__init__.py

```diff
@@ -347,13 +347,23 @@ def jit(
     if additional_transforms is None:
         additional_transforms = []
 
+    # Resolve names of executors
+    exc_list = executors if executors is not None else ()
+    executors = []
+    for e in exc_list:
+        if isinstance(e, str):
+            e = extend.get_executor(e)
+        if not isinstance(e, Executor):
+            raise TypeError(f"Expected an Executor or the name of one, instead got: {e}")
```

Comment:
This change is very welcome. I wrote my own version in https://github.com/Lightning-AI/litgpt/blob/main/extensions/thunder/strategies/utils.py#L7

It would be useful that you validate all the executors at once, as I do in that link. Having to rerun multiple times is inconvenient

Please also steal my test: https://github.com/Lightning-AI/litgpt/blob/main/tests/test_thunder_fsdp.py#L35-L44

Reply:
I added the test, and another to make sure it worked with custom executors. I also made `thunder.jit()` check if there were any duplicate entries in the executor list. Turns out we had a few bugs lurking.